hat/assets/js/apps/Iaso/components/forms/InputComponent.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { useState } from 'react';
+import { Box } from '@material-ui/core';
+import {
+    TextInput,
+    PasswordInput,
+    NumberInput,
+    Radio,
+    Checkbox,
+    ArrayFieldInput,
+    SearchInput,
+    // @ts-ignore
+    translateOptions,
+    Select,
+    useSafeIntl,
+} from 'bluesquare-components';
+import MESSAGES from '../../domains/forms/messages';
+
+export type InputComponentProps = {
+    type?: string;
+    keyValue: string;
+    value?: any;
+    errors?: string[];
+    // eslint-disable-next-line no-unused-vars
+    onChange?: (key: string, value: any) => void;
+    options?: any[];
+    disabled?: boolean;
+    multiline?: boolean;
+    clearable?: boolean;
+    label?: any;
+    labelString?: string;
+    required?: boolean;
+    onEnterPressed?: () => void;
+    withMarginTop?: boolean;
+    multi?: boolean;
+    uid?: any;
+    loading?: boolean;
+    getOptionLabel?: any;
+    getOptionSelected?: any;
+    renderOption?: any;
```

Comment:
These are all functions no?

From cursory glance, `renderOption` is a function component, `getOptionSelected` return a bool, and `getOptionsLabel` should return `string|undefined`

In the js component these props could also be an object (`Record<string,string>`) but I didn't see such use in the codebase